AdyenCard/Components/Card/CardViewController.swift

```diff
@@ -423,15 +423,33 @@ extension CardViewController {
     private func handleCardScanningResult(_ result: Result<CardScannerCardDetails, Error>) {
         switch result {
         case let .success((number, expiryDate)):
-            items.numberContainerItem.setCardNumber(number ?? "")
+            if let number {
+                items.numberContainerItem.setCardNumber(number)
+            }
+
             if let expiryDate {
                 items.expiryDateItem.setExpiryDate(expiryDate)
             }
+
+            focusNextInputField()
         case .failure:
-            // TODO: Implement error handling
-            break
+            cardScannerController.dismiss(nil)
         }
     }
+
+    private func focusNextInputField() {
+        let textInputItems: [FormTextInputItem] = [
```

Comment:
This is fragile as we redefine the order of fields the second time. Can we use our existing `CardViewController` form items and filter out them? Then if we have any redesign in the future we wouldn't need to remember updating this block.